lib/src/backend.rs

```diff
@@ -143,19 +144,52 @@ content_hash! {
     }
 }
 
+/// Identifies a single legacy tree, which may have path-level conflicts, or a
+/// merge of multiple trees, where the individual trees do not have conflicts.
+// TODO(#1624): Delete this type at some point in the future, when we decide to drop
+// support for conflicts in older repos, or maybe after we have provided an upgrade
+// mechanism.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum MergedTreeId {
+    /// The tree id of a legacy tree
+    Legacy(TreeId),
+    /// The tree id(s) of a merge tree
+    Merge(Merge<TreeId>),
+}
+
+impl ContentHash for MergedTreeId {
+    fn hash(&self, state: &mut impl Update) {
```

Comment:
nit: other impls specify it as `digest::Update` (maybe because `Update` is too generic?)

Reply:
Done